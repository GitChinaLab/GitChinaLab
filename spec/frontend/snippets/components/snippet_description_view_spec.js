import { shallowMount } from '@vue/test-utils';
import SnippetDescription from '~/snippets/components/snippet_description_view.vue';

describe('Snippet Description component', () => {
  let wrapper;
  const description = '<h2>The property of Thor</h2>';

  function createComponent() {
    wrapper = shallowMount(SnippetDescription, {
      propsData: {
        description,
      },
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
