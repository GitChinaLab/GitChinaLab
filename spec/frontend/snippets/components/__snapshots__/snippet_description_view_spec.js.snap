// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet Description component matches the snapshot 1`] = `
<markdown-field-view-stub
  class="snippet-description"
  data-qa-selector="snippet_description_content"
>
  <div
    class="md js-snippet-description"
  >
    <h2>
      The property of Thor
    </h2>
  </div>
</markdown-field-view-stub>
`;
